pandas/core/ops.py

```diff
@@ -663,6 +663,10 @@ def wrapper(left, right, name=name, na_op=na_op):
                     not isinstance(lvalues, pd.DatetimeIndex)):
                 lvalues = lvalues.values
 
+            if (hasattr(rvalues, 'values') and
+                    not isinstance(rvalues, pd.DatetimeIndex)):
```

Comment:
It looked to me as the safest choice. I wasn't sure if it was a job for _TimeOp and I didn't want to mess there.